File: P02 - Kelas Objek/Python/Bag. pertama/Kelas09.py
# File: Kelas08.py
# Penulis : Elmosius Suli
# Behavior / method & pembuatan objek

import datetime

class Kelas09:
    def __init__(self, age):
       self.__age = age
    
    def computeYob(self) -> int:    # method hitung tahun kelahiran
        now = datetime.datetime.now()
        return now.year - self.__age
    
    @property
    def age(self):          # accessor
        return self.__age

    
    @age.setter 
    def age(self, age:int): # mutator
        self.__age = age
        
## Program utama ##
def main():
    k = Kelas09(21)           # membuat objek k dari kelas09
        
    print("Age: ", k.age)     # berhasil, lewat accessor / getter
    print("Tahun lahir: ", k.computeYob())
    k.age = 17                # berhasil, lewat mutator / setter
    print("Age: ", k.age)    
    print("Tahun lahir: ", k.computeYob())
    
if __name__ == '__main__':
    main()

File: P02 - Kelas Objek/Java/Bag. pertama/Kelas09.java
package Praktikum.P02;

import java.time.LocalDate;

// Behaviout / method & pembuatan objek
public class Kelas09 {
    private int age;

    public Kelas09(int age){   // konstuktor
        this.age = age;
    }

    public int computeYob(){
        LocalDate now = LocalDate.now();
        return now.getYear() - this.age;
    }

    public int getAge(){      // accessor / getter method
        return this.age;
    }

    public void setAge(int age){    // mutator / setter method
        this.age = age;
    } 

    public static void main(String[] args){
        Kelas09 k;
        k = new Kelas09(21);     // membuat objek k dari kelas Kelas09
        System.out.println("Age: " + k.getAge());   // berhasil, lewat accesor / getter
        System.out.println("Tahun lahir: " + k.computeYob());   
        k.setAge(17);                           // berhasil, lewat mutator / setter
        System.out.println("Age: " + k.getAge());
        System.out.println("Tahun lahir: " + k.computeYob());   
    }



}
